src/object/obj_class.c

```diff
@@ -180,145 +212,22 @@ oclass_fit_max(daos_oclass_id_t oc_id, int domain_nr, int target_nr)
-
-	D_ASSERT(target_nr > 0);
-	D_ASSERT(domain_nr > 0);
-
-	oc = oclass_fit_max(oc_id, domain_nr, target_nr);
-	if (oc)
-		*oc_id_p = oc->oc_id;
-
-	return oc ? 0 : -DER_NONEXIST;
-}
-
-int
-dc_set_oclass(uint64_t rf_factor, int domain_nr, int target_nr,
-	      daos_ofeat_t ofeats, daos_oclass_hints_t hints,
-	      daos_oclass_id_t *oc_id_p)
-{
-	daos_oclass_id_t	cid = 0;
-	struct daos_obj_class	*oc;
-	struct daos_oclass_attr	ca;
-	uint16_t		shd, rdd;
-	int			grp_size;
-
-	rdd = hints & DAOS_OCH_RDD_MASK;
-	shd = hints & DAOS_OCH_SHD_MASK;
-
-	/** first set a reasonable default based on RF & RDD hint (if set) */
-	switch (rf_factor) {
-	case DAOS_PROP_CO_REDUN_RF0:
-		if (rdd == DAOS_OCH_RDD_RP) {
-			cid = OC_RP_2GX;
-		} else if (rdd == DAOS_OCH_RDD_EC) {
-			if (domain_nr >= 10)
-				cid = OC_EC_8P1GX;
-			else if (domain_nr >= 6)
-				cid = OC_EC_4P1GX;
-			else
-				cid = OC_EC_2P1GX;
-		} else {
-			cid = OC_SX;
-		}
-		break;
-	case DAOS_PROP_CO_REDUN_RF1:
-		if (rdd == DAOS_OCH_RDD_EC || ofeats & DAOS_OF_ARRAY ||
-		    ofeats & DAOS_OF_ARRAY_BYTE) {
-			if (domain_nr >= 10)
-				cid = OC_EC_8P1GX;
-			else if (domain_nr >= 6)
-				cid = OC_EC_4P1GX;
-			else
-				cid = OC_EC_2P1GX;
-		} else {
-			cid = OC_RP_2GX;
-		}
-		break;
-	case DAOS_PROP_CO_REDUN_RF2:
-		if (rdd == DAOS_OCH_RDD_EC || ofeats & DAOS_OF_ARRAY ||
-		    ofeats & DAOS_OF_ARRAY_BYTE) {
-			if (domain_nr >= 10)
-				cid = OC_EC_8P2GX;
-			else if (domain_nr >= 6)
-				cid = OC_EC_4P2GX;
-			else
-				cid = OC_EC_2P2GX;
-		} else {
-			cid = OC_RP_3GX;
-		}
-		break;
-	case DAOS_PROP_CO_REDUN_RF3:
-		/** EC not supported here */
-		cid = OC_RP_4GX;
-		break;
-	case DAOS_PROP_CO_REDUN_RF4:
-		/** EC not supported here */
-		cid = OC_RP_6GX;
-		break;
-	}
-
-	/** we have determined the resilience part, now set the grp size */
-
-	oc = oclass_ident2cl(cid);
-	if (!oc)
-		return -DER_INVAL;
-
-	memcpy(&ca, &oc->oc_attr, sizeof(ca));
-	grp_size = daos_oclass_grp_size(&ca);
-
-	/** adjust the group size based on the sharding hint */
-	switch (shd) {
-	case 0:
-	case DAOS_OCH_SHD_DEF:
-		if (ofeats & DAOS_OF_ARRAY || ofeats & DAOS_OF_ARRAY_BYTE ||
-		    ofeats & DAOS_OF_KV_FLAT)
-			ca.ca_grp_nr = DAOS_OBJ_GRP_MAX;
-		else
-			ca.ca_grp_nr = 1;
-		break;
-	case DAOS_OCH_SHD_MAX:
-		ca.ca_grp_nr = DAOS_OBJ_GRP_MAX;
-		break;
-	case DAOS_OCH_SHD_TINY:
-		ca.ca_grp_nr = 4;
-		break;
-	case DAOS_OCH_SHD_REG:
-		ca.ca_grp_nr = max(128, target_nr * 25 / 100);
-		break;
-	case DAOS_OCH_SHD_HI:
-		ca.ca_grp_nr = max(256, target_nr * 50 / 100);
-		break;
-	case DAOS_OCH_SHD_EXT:
-		ca.ca_grp_nr = max(1024, target_nr * 80 / 100);
-		break;
-	default:
-		D_ERROR("Invalid sharding hint\n");
-		return -DER_INVAL;
-	}
-
-	if (ca.ca_grp_nr == DAOS_OBJ_GRP_MAX ||
-	    ca.ca_grp_nr * grp_size > target_nr) {
-		/* search for the highest scalability in the allowed range */
-		ca.ca_grp_nr = max(1, (target_nr / grp_size));
+	if (oc_id < (OR_RP_1 << OC_REDUN_SHIFT)) {
+		*ord = 0;
```

Comment:
what does "0" mean?

Reply:
This is a bit tricky, *ord = 0 means this is special object class that is less than 50. and nr_groups will be equal as class.